---
title: "Genomic distribution of genes potentially regulated by Kar4"
output: html_notebook
---

### Plot intergenic transcripts dispersion across chromsomes
```{r}
library(karyoploteR)
library(GenomicRanges)

pdf(file = "./Plot_Jason/kar4_dep_up_final.pdf")

  pp <- getDefaultPlotParams(plot.type=2)
  #pp$data2inmargin <- 140
  
  # Load Sc genome
  kp <- karyoploteR::plotKaryotype(genome = "sacCer3", plot.type=1, plot.params = pp)
  
  # Add chromosome coords
  kpAddBaseNumbers(kp, tick.dist = 100000, units  = "kb", 
                   tick.len = 20, tick.col="red3", cex=0.5,
                   minor.tick.dist = 10000, minor.tick.len = 10, 
                   minor.tick.col = "gray2")
  
  
  library(rtracklayer)
  
  # Load list of genes of interest
  goi_1 <- read.table(file = "./data/kar4_dep_up_final.txt", header = FALSE)
  colnames(goi_1) <- "gene_id"
  
  # Load intergenic transcripts as genes
  features <- import(con = "./data/Saccharomyces_cerevisiae.R64-1-1.106_sorted_karyoploteR.gtf")
  
  #features_goi_1 <- subset(features, gene_id %in% goi_1$gene_id)
  features_goi_1 <- subset(features, gene_id %in% goi_1$gene_id)
  features_goi_1_plus <- subset(features_goi_1, strand == "+")
  features_goi_1_minus <- subset(features_goi_1, strand == "-")
  
  
  #genes_1 <- features[features_goi_1$type=="gene"]
  genes_1_plus <- features_goi_1_plus[features_goi_1_plus$type=="gene"]
  genes_1_minus <- features_goi_1_minus[features_goi_1_minus$type=="gene"]
  kpPlotRegions(kp, data=genes_1_plus, col = "green4", data.panel = 1, 
                r0=0.55, r1=1)
  kpPlotRegions(kp, data=genes_1_minus, col = "red3", data.panel = 1, 
                r0=0, r1=0.45, avoid.overlapping = FALSE)
  
  
  # Load centromere coordinates
  features <- import(con = "./data/centromeres.gff")
  centromeres <- features[features$type=="centromere"]
  kpRect(kp, data=centromeres, y0=0, y1=1, col = "black", data.panel="ideogram", border="black")
   
dev.off()
```

### Plot intergenic transcripts dispersion across chromsomes for kar4_only_up_final.txt
```{r}
pdf(file = "./Plot_Jason/kar4_only_up_final.pdf")

  pp <- getDefaultPlotParams(plot.type=2)
  #pp$data2inmargin <- 140
  
  # Load Sc genome
  kp <- karyoploteR::plotKaryotype(genome = "sacCer3", plot.type=1, plot.params = pp)
  
  # Add chromosome coords
  kpAddBaseNumbers(kp, tick.dist = 100000, units  = "kb", 
                   tick.len = 20, tick.col="red3", cex=0.5,
                   minor.tick.dist = 10000, minor.tick.len = 10, 
                   minor.tick.col = "gray2")
  
  
  library(rtracklayer)
  
  # Load list of genes of interest
  goi_2 <- read.table(file = "./data/kar4_only_up_final.txt", header = FALSE)
  colnames(goi_2) <- "gene_id"
  
  # Load intergenic transcripts as genes
  features <- import(con = "./data/Saccharomyces_cerevisiae.R64-1-1.106_sorted_karyoploteR.gtf")
  
  #features_goi_1 <- subset(features, gene_id %in% goi_1$gene_id)
  features_goi_2 <- subset(features, gene_id %in% goi_2$gene_id)
  features_goi_2_plus <- subset(features_goi_2, strand == "+")
  features_goi_2_minus <- subset(features_goi_2, strand == "-")
  
  
  #genes_1 <- features[features_goi_1$type=="gene"]
  genes_2_plus <- features_goi_2_plus[features_goi_2_plus$type=="gene"]
  genes_2_minus <- features_goi_2_minus[features_goi_2_minus$type=="gene"]
  kpPlotRegions(kp, data=genes_2_plus, col = "green4", data.panel = 1, 
                r0=0.55, r1=1)
  kpPlotRegions(kp, data=genes_2_minus, col = "red3", data.panel = 1, 
                r0=0, r1=0.45, avoid.overlapping = FALSE)
  
  
  # Load centromere coordinates
  features <- import(con = "./data/centromeres.gff")
  centromeres <- features[features$type=="centromere"]
  kpRect(kp, data=centromeres, y0=0, y1=1, col = "black", data.panel="ideogram", border="black")
   
dev.off()
```
